src/Symfony/Component/HttpClient/MockHttpClient.php

```diff
@@ -54,7 +61,10 @@ public function __construct($responseFactory = null, string $baseUri = null)
      */
     public function request(string $method, string $url, array $options = []): ResponseInterface
     {
-        [$url, $options] = $this->prepareRequest($method, $url, $options, ['base_uri' => $this->baseUri], true);
+        $defaultOptions = $this->defaultOptions;
+        $defaultOptions['base_uri'] = $this->baseUri;
```

Comment:
I'd suggest removing the baseUri property and setting the `base_uri` in $this->defaultOption in the constructor instead (when not null)

Reply:
Thanks, fixed.